### ISSUE TITLE
Pop!_Shop update changed my login screen to a broken cosmic login 

### ISSUE THREAD

**Issue by archaistvolts** [pop-os/gnome-shell-extension-pop-cosmic#413]
Latest update caused my login screen to change to what appears to be a new, broken cosmic login screen.   It had the 3 DE options I'm used to: gdm, gdm wayland and cosmic. However I wasn't able to login with either gdm option, only cosmic. I use gdm wayland daily and had only installed cosmic DE for a test drive. 

I managed to restore with timeshift. But now I don't really know what went wrong or how to proceed. Do I need to uninstall cosmic DE? I tried running this command to do that but it didn't work.

```console
$ sudo apt remove pop-cosmic libpop-desktop-widget gnome-shell-extension-cosmic-dock gnome-shell-extension-cosmic-workspaces
[sudo] password for travis: 
Reading package lists... Done
Building dependency tree... Done
Reading state information... Done
Some packages could not be installed. This may mean that you have
requested an impossible situation or if you are using the unstable
distribution that some required packages have not yet been created
or been moved out of Incoming.
The following information may help to resolve the situation:

The following packages have unmet dependencies:
 pop-desktop : PreDepends: pop-de-gnome but it is not going to be installed
E: Error, pkgProblemResolver::Resolve generated breaks, this may be caused by held packages.
```

Any suggestions on what might be wrong? I want to continue using gdm wayland and be able to update my system.

**Comment by archaistvolts** [comment]
I also posted here https://www.reddit.com/r/pop_os/comments/1ci8fo4/pop_shop_update_changed_my_login_screen_to_a/

**Comment by leviport** [comment]
As several responses have said in that Reddit thread already, set your DM back to GDM3 with `sudo dpkg-reconfigure gdm3`. Cosmic-greeter works great as a lock screen, but it still needs work as a DM, and that is currently known.

**Comment by archaistvolts** [comment]
Thank you @leviport 

Any idea why Pop!_Shop would have installed Cosmic-greeter as DM login then?  And is there some package I can `apt remove` or something else to change to prevent this so that gdm remains the DM login and not Cosmic-greeter when I update?  I am wary of updating now and would rather not have to worry about whether `sudo dpkg-reconfigure gdm3` will work.

**Comment by leviport** [comment]
Cosmic-greeter is a dependency of cosmic-session, so it probably showed up when you installed COSMIC DE. If you install it with the terminal, there is a prompt that appears and asks whether to use GDM3 or cosmic-greeter. I'm not sure whether cosmic-session is installable from Pop Shop, but if it is, then that prompt won't be able to appear. Since COSMIC DE is still a mix of alpha and pre-alpha, things like this can occur while testing it.

**Comment by archaistvolts** [comment]
Ah that makes more sense.  Pretty sure I installed cosmic DE from the command line but don't remember if I chose a DE option... its been a long time and not sure exactly when I installed - maybe 6 months or a year ago.  However the update was done with pop shop earlier this week and I'm sure I didn't choose an option.  So if cosmic-session was installed/updated with pop shop, that would explain why this happened.
